playground/frontend/playground_components/lib/src/controllers/snippet_editing_controller.dart

```diff
@@ -45,16 +47,26 @@ class SnippetEditingController extends ChangeNotifier {
     _onSymbolsNotifierChanged();
   }
 
-  set selectedExample(Example? value) {
-    _selectedExample = value;
+  void configure({
+    required Example? example,
+    ExampleLoadingDescriptor? descriptor,
+    String? pipelineOptions,
+  }) {
+    _descriptor = descriptor ?? _descriptor;
+    _selectedExample = example ?? _selectedExample;
+    
+    if (example != null && descriptor == null) {
+      _descriptor = null;
+    }
```

Comment:
This adds complexity with conditions.
I suggest adding `{required ExampleLoadingDescriptor descriptor}` and always overwrite both.

Mixing `pipelineOptions` here is extra confusing because `Example` itself contains `pipelineOptions`. So if we set both, what is the reason to prefer one `pipelineOptions` over another?

Reply:
fixed